### PR TITLE
[T3A3][T11-A2]Jordan May

### PR DESCRIPTION
**Pull request by jjamay**
Encapsulate each component of Address into separate classes

### CODE REVIEW

**Comment by jjamay** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -9,13 +12,62 @@
...
-    public final String value;
     private boolean isPrivate;
+    
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
 
     /**
+     * Encapsulates one piece of address information
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:27]: A super class header comment should give more information than this. It should be clear to a developer who is extending your superclass at any point in time.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -9,13 +12,62 @@
...
+        AddressItem(String value) {
+            this.value = value;
+        }
+        
+        public String getValue() {
+            return this.value;
+        }
+    }
+    
+    private class Block extends AddressItem {
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:43]: Non-trivial classes, even if they are nested should have header comments.

File: test/expected.txt [hunk 3]
```diff
@@ -8,6 +8,7 @@
 || Enter command: || [Command entered:  sfdfd]
 || add: Adds a person to the address book. Contact details can be marked private by prepending 'p' to the prefix.
 || Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...
+|| ADDRESS should be entered as a/BLOCK, STREET, UNIT, POSTAL_CODE
```
Comment (nirandiw) [inline test/expected.txt:11]: Good that you updated the tests. 

Next time update the user guide as well.

**Review by nirandiw — COMMENTED** [review]
@jjamay Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**